omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/TaskActionController1_8Test.java

```diff
@@ -0,0 +1,145 @@
+	
+	@Before
+	public void setUp() throws Exception {
+		mockTaskServiceWrapper.registeredTasks.addAll(Arrays.asList(testTask, tempTask));
+		
+		TaskActionResource1_8 taskActionResource = (TaskActionResource1_8) restService
+		        .getResourceBySupportedClass(TaskAction.class);
+		taskActionResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+		
+		TaskDefinitionResource1_8 taskResource = (TaskDefinitionResource1_8) restService
+		        .getResourceBySupportedClass(TaskDefinition.class);
+		taskResource.setTaskServiceWrapper(mockTaskServiceWrapper);
+	}
+	
+	@Test
+	public void shouldScheduleTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+	}
+	
+	@Test
+	public void shouldShutdownTask() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask);
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTaskAlreadyScheduled() throws Exception {
+		mockTaskServiceWrapper.scheduledTasks.add(testTask);
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"scheduletask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, hasItem(testTask));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+	}
+	
+	@Test
+	public void shouldDoNothingIfTaskAlreadyShutdown() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"shutdowntask\", \"tasks\":[\"" + getTestTaskName()
+		        + "\"]}")));
+		//check if state preserved
+		assertThat(mockTaskServiceWrapper.scheduledTasks, not(hasItem(testTask)));
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+	}
+	
+	@Test
+	public void shouldDeleteTask() throws Exception {
+		//sanity check
+		assertThat(mockTaskServiceWrapper.registeredTasks, hasItem(testTask));
+		deserialize(handle(newPostRequest(getURI(), "{\"action\": \"delete\", \"tasks\":[\"" + getTestTaskName() + "\"]}")));
+		assertThat(mockTaskServiceWrapper.registeredTasks, not(hasItem(testTask)));
+	}
+	
+	@Override
+	@Test(expected = Exception.class)
+	public void shouldGetDefaultByUuid() throws Exception {
```

Comment:
should get what?

Reply:
Overridden method. It is also like used to get Default Representation. But not supported to TaskAction.